.github/workflows/pypi-package.yml

```diff
@@ -45,3 +48,5 @@ jobs:
 
       - name: Upload package to PyPI
         uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          attestations: true
```

Comment:
So this is what actually generates the attestations, and even though the feature is still experimental we can still enable it here (and in the other repos) and let `gh-action-pypi-publish` deal with potential API changes, right?

I guess the other changes are unrelated, but it was convenient to bundle them together.